hazelcast/src/main/java/com/hazelcast/jet/impl/operation/TerminateJobOperation.java

```diff
@@ -52,7 +60,10 @@ public CompletableFuture<Void> doRun() {
             getJobCoordinationService().terminateLightJob(jobId(), true);
             return completedFuture(null);
         } else {
-            return getJobCoordinationService().terminateJob(jobId(), terminationMode, true);
+            JobCoordinationService jobCoordinationService = getJobCoordinationService();
+            return checkJobIsSuspendable(jobCoordinationService)
```

Comment:
this is wrong place, see https://github.com/hazelcast/hazelcast/pull/25939#discussion_r1386997354

Reply:
I tried it on the upper level, but it doesn't work as I've expected. I decided that here is the best place for that,

Reply:
it does not prevent suspension in all cases

Reply:
Need to prepare a test to understand how exactly it is possible...

Reply:
Will be tested in EE : https://github.com/hazelcast/hazelcast-enterprise/pull/6812